shared-module/atexit/__init__.c

```diff
@@ -79,11 +74,18 @@ void shared_module_atexit_unregister(const mp_obj_t *func) {
     }
 }
 
-void shared_module_atexit_execute(void) {
+void shared_module_atexit_execute(pyexec_result_t *result) {
     if (callback) {
         for (size_t i = callback_len; i-- > 0;) {
             if (callback[i].func != mp_const_none) {
-                mp_call_function_n_kw(callback[i].func, callback[i].n_pos, callback[i].n_kw, callback[i].args);
+                if (result != NULL) {
+                    pyexec_result_t res;
+                    if (pyexec_exit_handler(&callback[i], &res) == PYEXEC_DEEP_SLEEP) {
+                        *result = res;
+                    }
+                } else {
+                    pyexec_exit_handler(&callback[i], NULL);
+                }
```

Comment:
did you consider using nlr_push to allow catching exceptions that originate in the exit handlers? It looks like you had your reasons, but it would be helpful to record them.

Reply:
I didn't find any effective use case for recording them, do let me know of what you have in mind.